packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1156,11 +1201,13 @@ export class TransactionController extends BaseController<
         const cancelError = rpcErrors.internal(
           'User cancelled the transaction',
         );
-
         resultCallbacks?.error(cancelError);
         throw cancelError;
 
       case TransactionStatus.submitted:
+        this.hub.emit(TransactionEvent.submitted, {
```

Comment:
Rather than firing this event at the end of the promise, should we instead emit it in inline inside `approveTransaction` after we call `sendRawTransaction`?

Reply:
This is done in https://github.com/MetaMask/core/pull/1894/commits/02906c6be2bc0271a23b83c7fd8db244f24a591f